frontend/src/lib/MlmdUtils.test.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  Api,
+  Context,
+  GetContextByTypeAndNameRequest,
+  GetContextByTypeAndNameResponse,
+} from '@kubeflow/frontend';
+import { expectWarnings, testBestPractices } from 'src/TestUtils';
+import { Workflow, WorkflowSpec, WorkflowStatus } from 'third_party/argo-ui/argo_template';
+import { getRunContext } from './MlmdUtils';
+
+testBestPractices();
+
+const WORKFLOW_NAME = 'run-st448';
+const WORKFLOW_EMPTY: Workflow = {
+  metadata: {
+    name: WORKFLOW_NAME,
+  },
+  // there are many unrelated fields here, omit them
+  spec: {} as WorkflowSpec,
+  status: {} as WorkflowStatus,
+};
+
+const V2_CONTEXT = new Context();
+V2_CONTEXT.setName(WORKFLOW_NAME);
+V2_CONTEXT.setType('kfp.PipelineRun');
+
+const TFX_CONTEXT = new Context();
+TFX_CONTEXT.setName('run.run-st448');
+TFX_CONTEXT.setType('run');
+
+const V1_CONTEXT = new Context();
+V1_CONTEXT.setName(WORKFLOW_NAME);
+V1_CONTEXT.setType('KfpRun');
+
+describe('MlmdUtils', () => {
+  describe('getRunContext', () => {
+    it('gets KFP v2 context', async () => {
+      mockGetContextByTypeAndName([V2_CONTEXT]);
+      const context = await getRunContext({
+        ...WORKFLOW_EMPTY,
+        metadata: {
+          ...WORKFLOW_EMPTY.metadata,
+          annotations: { 'pipelines.kubeflow.org/v2_pipeline': 'true' },
+        },
+      });
+      expect(context).toEqual(V2_CONTEXT);
+    });
+
+    it('gets TFX context', async () => {
+      mockGetContextByTypeAndName([TFX_CONTEXT, V1_CONTEXT]);
+      const context = await getRunContext(WORKFLOW_EMPTY);
+      expect(context).toEqual(TFX_CONTEXT);
+    });
+
+    it('gets KFP v1 context', async () => {
+      const verify = expectWarnings();
```

Comment:
It might not be necessary to define a const value `verify`. We can call expectWarnings() directly when needed, which is increase readability.

Reply:
Note, expectWarnings() need two steps:
1. set up jest spy on logger.warn method
2. verify this test actually called warnings

so it had to be two steps

Reply:
I see, I understand now the return value of `expectWarnings()` is for verification.